src/test/parse-fail/issue-33569.rs

```diff
@@ -0,0 +1,15 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// compile-flags: -Z no-analysis
+
+macro_rules! foo {
+  { $+ } => () //~ ERROR: expected identifier, found `+`
+}              //~^ ERROR: no rules expected the token `}`
```

Comment:
If we removed the call to `self.bump()` above, I believe the second error would be
```
no rules expected for the token `+`
```

I would prefer this error since there's nothing wrong with the `}` token.